src/transformers/adapters/models/electra/adapter_model.py

```diff
@@ -0,0 +1,236 @@
+            optional): Activation function. Defaults to 'tanh'. overwrite_ok (bool, optional): Force overwrite if a
+            head with the same name exists. Defaults to False.
+        """
+        head = TaggingHead(self, head_name, num_labels, layers, activation_function, id2label)
+        self.add_prediction_head(head, overwrite_ok)
+
+    def add_qa_head(
+        self, head_name, num_labels=2, layers=1, activation_function="tanh", overwrite_ok=False, id2label=None
+    ):
+        head = QuestionAnsweringHead(self, head_name, num_labels, layers, activation_function, id2label)
+        self.add_prediction_head(head, overwrite_ok)
+
+    def add_dependency_parsing_head(self, head_name, num_labels=2, overwrite_ok=False, id2label=None):
+        """
+        Args:
+        Adds a biaffine dependency parsing head on top of the model. The parsing head uses the architecture described
+        in "Is Supervised Syntactic Parsing Beneficial for Language Understanding? An Empirical Investigation" (Glavaš
+        & Vulić, 2021) (https://arxiv.org/pdf/2008.06788.pdf).
+            head_name (str): The name of the head. num_labels (int, optional): Number of labels. Defaults to 2.
+            overwrite_ok (bool, optional): Force overwrite if a head with the same name exists. Defaults to False.
+            id2label (dict, optional): Mapping from label ids to labels. Defaults to None.
+        """
+        head = BiaffineParsingHead(self, head_name, num_labels, id2label)
+        self.add_prediction_head(head, overwrite_ok)
+
+    def add_masked_lm_head(self, head_name, activation_function="gelu", overwrite_ok=False):
+        """
+        Args:
+        Adds a masked language modeling head on top of the model.
+            head_name (str): The name of the head. activation_function (str, optional): Activation function. Defaults
+            to 'gelu'. overwrite_ok (bool, optional): Force overwrite if a head with the same name exists. Defaults to
+            False.
+        """
+        head = BertStyleMaskedLMHead(self, head_name, activation_function=activation_function)
+        self.add_prediction_head(head, overwrite_ok=overwrite_ok)
+
+    def add_causal_lm_head(self, head_name, activation_function="gelu", overwrite_ok=False):
+        """
+        Args:
+        Adds a causal language modeling head on top of the model.
+            head_name (str): The name of the head. activation_function (str, optional): Activation function. Defaults
+            to 'gelu'. overwrite_ok (bool, optional): Force overwrite if a head with the same name exists. Defaults to
+            False.
+        """
+        head = CausalLMHead(
+            self, head_name, layers=2, activation_function=activation_function, layer_norm=True, bias=True
+        )
+        self.add_prediction_head(head, overwrite_ok=overwrite_ok)
+
+
+class ElectraModelWithHeads(ElectraAdapterModel):
```

Comment:
The model classes of the form XModelWithHeads are deprecated, so we don't want to add those classes for newly supported architectures. `ElectraAdapterModel` should be used for all cases. Please remove this class.